### DISCUSSION TITLE
how to add new keymap / actions without removing the default one

### DISCUSSION THREAD

**Discussion by WilliamHsieh** [ibhagwan/fzf-lua#1327] [Q&A] [answered]
When I add a keymap or action in my current setup, the other default keymap/action will not work.
A simple [workaround](https://github.com/WilliamHsieh/dotfiles/blob/2c7b5301d84422ad5c9a547eb9568b079dcd0135/config/nvim/lua/plugins/fzf.lua) is to copy all the default config from readme along with my added ones.
Is there a better way to achieve this?

**Comment by zhu-jiyuan** [comment]
![image](https://github.com/user-attachments/assets/84949ff4-e358-4fa7-b9d2-ea1a1ce734c6)
You can use keys or vim.keymap.set.

**Reply by ibhagwan** [reply]
As mentioned in the comments just above these tables the keymap tables don’t inherit the defaults, you can manually merge the defaults or I can add an option to copy the defaults.

**Reply by ibhagwan** [reply, marked as answer]
Update fzf-lua, post https://github.com/ibhagwan/fzf-lua/commit/bd291dc51581a49bee50ea7a309fad7137f7e378 you can now do:
```lua
```lua
reqire("fzf-lua").setup({
  keymap = {
    fzf = {
      true,  -- inherit defaults
      ["ctrl-q"] = "select-all+accept"
    }
  },
})
```

**Reply by WilliamHsieh** [reply]
thank you so much!
